src/main/java/org/graylog2/syslog4j/impl/unix/socket/UnixSocketSyslog.java

```diff
@@ -39,6 +40,12 @@ public void setSunPath(String sunPath) {
             System.arraycopy(sunPath.getBytes(), 0, this.sun_path, 0, sunPath.length());
             System.arraycopy(ZERO_BYTE, 0, this.sun_path, sunPath.length(), 1);
         }
+
+		@Override
+		protected List getFieldOrder() {

```

Comment:
I should by the way thank you for this syslog4j as it is quite useful :) I
have now pushed the fix for avoiding NPE possibility.

On Thu, Feb 18, 2016 at 11:47 AM, Negar Safinia <negar7918@gmail.com> wrote:

> I checked the older version of jna and they had implementation for the
> method but I used a newer version and it is abstract:
>
>
> http://grepcode.com/file/repo1.maven.org/maven2/net.java.dev.jna/jna/3.5.0/com/sun/jna/Structure.java#Structure.getFieldOrder%28%29
>
> But sure to avoid NPE I can return an empty list (It is not used now so I
> didn't care really)
>
> On Thu, Feb 18, 2016 at 11:33 AM, Jochen Schalanda <
> notifications@github.com> wrote:
>
>> In
>> src/main/java/org/graylog2/syslog4j/impl/unix/socket/UnixSocketSyslog.java
>> <https://github.com/Graylog2/syslog4j-graylog2/pull/6#discussion_r53296629>
>> :
>>
>> > @@ -39,6 +40,12 @@ public void setSunPath(String sunPath) {
>> >              System.arraycopy(sunPath.getBytes(), 0, this.sun_path, 0, sunPath.length());
>> >              System.arraycopy(ZERO_BYTE, 0, this.sun_path, sunPath.length(), 1);
>> >          }
>> > +
>> > +		@Override
>> > +		protected List getFieldOrder() {
>>
>> The Javadoc at
>> https://jna.java.net/nonav/javadoc/com/sun/jna/Structure.html#getFieldOrder()
>> doesn't mention this method being abstract and "my" JNA implementation
>> provides a standard implementation for this method.
>>
>> The question is, if is has to be overridden (it hasn't been overridden
>> before this PR), and if it has to, whether it should return null (which
>> might cause NPEs across the board) or simply return an empty list.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Graylog2/syslog4j-graylog2/pull/6/files#r53296629>.
>>
>
>